### ISSUE TITLE
spaces before attribute access not reported as violation

### ISSUE THREAD

**Issue by wbolster** [PyCQA/pycodestyle#553]
Code like this:

```python
d = {}
d     .items()
````

should result in a warning about excessive whitespace.

**Comment by sigmavirus24** [comment]
I agree that I'd like this but I don't see mention of that in [PEP 8](https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements). This should probably be off-by-default as a result of it not being called out specifically in the document.

**Comment by wbolster** [comment]
it's similar to [] indexing/slicing, which also gets no spaces before it.

    — wouter

-- 
sent from my phone. please forgive my tpyos.

**Comment by sigmavirus24** [comment]
@wbolster I'm not sure if @IanLee1521 wants to expand that to attribute access. But I agree with you entirely and I'm happy to write this check.

**Comment by IanLee1521** [comment]
Wow... I would have expected that to be a syntax error.

I agree that we should add the check, disabled by default (for now), but I'd like to see PEP-8 updated as well, I submitted a PR at https://github.com/python/peps/pull/36 for that change.

**Comment by wbolster** [comment]
it's not a syntax error, since long chained attribute accesses or chained method calls can be split over multiple lines. on a single line, whitespace before a dot is just ugly. hence my style check request. :)

    — wouter

-- 
sent from my phone. please forgive my tpyos.

**Comment by IanLee1521** [comment]
Yeah, as I dug into it I realized the multi-line chaining was the reason for the lack of `SyntaxError`. :)

**Comment by anntzer** [comment]
Just as a an additional point, spaces *after* the dot (`foo. bar`) currently do not cause any warning either.
